### PR TITLE
[ttLib] Add main to ttFont and ttCollection

### PR DESCRIPTION
**Pull request by behdad**
Just opens font files from commandline.

### CODE REVIEW

**Comment by behdad** [comment]
I can add the docstrings and help stuff to show up in "./fonttools --help" as well. Let me do that.

**Comment by justvanrossum** [comment]
Why is this this useful and why should these modules become subpackages?

**Comment by behdad** [comment]
> Why is this this useful

Helps debugging errors in opening fonts. A niche for sure but I find myself firing up Python console to just TTFont() files often. I'm happy to leave them unlisted.

> and why should these modules become subpackages?

If I add `__name__ == "__main__"` in `ttFont.py` it hits include-order issue because ttFont is first included by ttLib; I'll get a warning from Python.

**Comment by behdad** [comment]
(I was debugging https://github.com/fonttools/fonttools/issues/1362 when I had this idea.)

**Comment by behdad** [comment]
Another usecase I was dealing with today was to measure memory consumption of opening fonts.

The modules can also take `--[no-]lazy`, `--save`, etc. as well.

**Comment by anthrotype** [comment]
hm I'm not convinced this should be added, these CLI entry points don't do anything useful for the end-user, and a developer can always write a three line script to test stuff

**Comment by khaledhosny** [comment]
Opening and saving a font with FontTools is a good optimization trick, currently one has to do it the slow way using ttx twice or write a script. I think it is worthwhile to have these CLI entries.

**Comment by anthrotype** [comment]
well.. if at least two of you find these useful, I'm OK to add them! Make it optionally save and then it becomes more useful like Khaled suggested

**Comment by justvanrossum** [comment]
`ttLib` itself doesn't have a `__main__`, perhaps these ideas could fit there? I would be a less intrusive change.

**Comment by behdad** [comment]
> `ttLib` itself doesn't have a `__main__`, perhaps these ideas could fit there? It would be a less intrusive change.

Currently it is non-trivial to distinguish TTC from non-TTC without peeking at the font bytes... I can make ttLib main just do TTFont for now...

**Comment by behdad** [comment]
> Currently it is non-trivial to distinguish TTC from non-TTC without peeking at the font bytes... I can make ttLib main just do TTFont for now...

Or have a `--collection`

**Comment by behdad** [comment]
Or I can return a specific TTLibError subclass and catch it. Let me do that.

**Comment by behdad** [comment]
Okay, simplified it.

Will look into save, --font-number, etc. It can be used to extract a font from a collection, and possibly to combine fonts into a collection as well.

**Comment by behdad** [comment]
PTAL! This is ready except for docstring.

**Comment by behdad** [comment]
I wrote a rudimentary docstring, but `--help` is not implemented. I suggest merging for now?

File: Lib/fontTools/ttLib/__init__.py [hunk 1]
```diff
@@ -15,3 +17,61 @@ def debugmsg(msg):
...
 from fontTools.ttLib.ttFont import *
 from fontTools.ttLib.ttCollection import TTCollection
+
+
+def main(args=None):
+	"""Open font with TTFont() or TTCollection()
+
+    ./fonttools ttLib [-oFILE] [-yNUMBER] files...
+
+	If multiple files are give on the command-line,
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/__init__.py:27]:
```suggestion
	If multiple files are given on the command-line,
```

File: Lib/fontTools/ttLib/__init__.py [hunk 2]
```diff
@@ -15,3 +17,70 @@ def debugmsg(msg):
 
 from fontTools.ttLib.ttFont import *
 from fontTools.ttLib.ttCollection import TTCollection
+
+
+def main(args=None):
+	"""Open font with TTFont() or TTCollection()
+
+    ./fonttools ttLib [-oFILE] [-yNUMBER] files...
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/__init__.py:25]: the indentation seems off

File: Lib/fontTools/ttLib/__init__.py [hunk 3]
```diff
@@ -15,3 +17,70 @@ def debugmsg(msg):
...
+    If --lazy or --no-lazy are give, those are passed
+    to the TTFont() or TTCollection() constructors.
+	"""
+	from fontTools import configLogger
+
+	if args is None:
+		args = sys.argv[1:]
+
+	import getopt
+	options, files = getopt.getopt(args, "o:y:",
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/__init__.py:52]: i'd have expected to use argparse instead of getopt, but it's ok

Comment (justvanrossum) [reply Lib/fontTools/ttLib/__init__.py:52]: argparse is so much better. The Python docs for getopt start with this highlighted box:

> Note The [getopt](https://docs.python.org/3/library/getopt.html#module-getopt) module is a parser for command line options whose API is designed to be familiar to users of the C getopt() function. Users who are unfamiliar with the C getopt() function or who would like to write less code and get better help and error messages should consider using the [argparse](https://docs.python.org/3/library/argparse.html#module-argparse) module instead.

I think fonttools should use argparse for all new code.

**Comment by justvanrossum** [comment]
> Currently it is non-trivial to distinguish TTC from non-TTC without peeking at the font bytes

Looking at the file name extension is better than nothing. I've only ever seen `.ttc` and `.otc`.

**Comment by behdad** [comment]
Ported to argparse. PTAL!

**Review by anthrotype — APPROVED** [review]
LGTM

**Pull request merged**